x-pack/plugins/security_solution/public/detection_engine/rule_management_ui/components/rules_table/rules_table/use_clear_rules_table_saved_state.ts

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useCallback } from 'react';
+import { useReplaceUrlParams } from '../../../../../common/utils/global_query_string/helpers';
+import { useKibana } from '../../../../../common/lib/kibana';
+import { URL_PARAM_KEY } from '../../../../../common/hooks/use_url_state';
+import { RULES_TABLE_STATE_STORAGE_KEY } from '../constants';
+import { useRulesTableContext } from './rules_table_context';
+import {
+  DEFAULT_FILTER_OPTIONS,
+  DEFAULT_PAGE,
+  DEFAULT_RULES_PER_PAGE,
+  DEFAULT_SORTING_OPTIONS,
+} from './rules_table_defaults';
+
+export function useClearRulesTableSavedState(): () => void {
```

Comment:
Let's move it to the `RulesTableContext` -- its actions are lacking high-level methods with domain logic, which makes it harder to use in the code, test, and understand how the table works.

Reply:
Agree, it will be much clear this way.